src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +273,485 @@ define(function (require, exports, module) {
+                    if (MainViewManager.findInWorkingSet("first-pane", prefsPath) >= 0) {
+
+                        MainViewManager._moveView("first-pane", "second-pane", file, 0, true);
+
+                        // Now refresh the project tree by asking
+                        // it to rebuild the UI.
+                        WorkingSetView.refresh(true);
+                    }
+
+                    CommandManager.execute(Commands.FILE_OPEN, { fullPath: prefsPath, paneId: "second-pane"})
+                        .done(function () {
+                            deferredPromise.resolve();
+                        }).fail(function () {
+                            deferredPromise.reject();
+                        });
+                }).fail(function () {
+                    deferredPromise.reject();
+                });
+        }
+
+        var resultObj = MainViewManager.findInAllWorkingSets(defaultPrefsPath);
+        if (resultObj && resultObj.length > 0) {
+            CommandManager.execute(Commands.FILE_CLOSE, {file: defaultPrefsFile, paneId: resultObj[0].paneId})
+                .done(function () {
+                    _openFiles();
+                }).fail(function () {
+                    deferredPromise.reject();
+                });
+        } else {
+            _openFiles();
+        }
+
+    }
+
+    function _isSupportedPrefType(prefType) {
+        if (SUPPORTED_PREFERENCE_TYPES.indexOf(prefType) >= 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+   /*
+    * This method tries to deduce the preference type
+    * based on various parameters like objects initial
+    * value, object type, object's type property.
+    */
+    function _getObjType(prefObj) {
+
+        var prefType = "undefined";
+
+        if (prefObj) {
+            // check the type parameter.
+            var _type = prefObj.type;
+            if (_type !== undefined) {
+                prefType = prefObj.type.toLowerCase();
+                // make sure the initial property's
+                // object type matches to that of 'type' propety.
+                if (prefObj.initial !== undefined) {
+
+                    if (Array.isArray(prefObj.initial)) {
+                        _type = "array";
+                    } else {
+                        var _initialType = typeof (prefObj.initial);
+                        _initialType = _initialType.toLowerCase();
+                        if (_type !== _initialType) {
+                            _type = _initialType;
+                        }
+                    }
+                }
+            }
+
+            if (_type) {
+                // preference object's type
+                // is defined. Check if that is valid or not.
+                prefType = _type;
+                if (!_isSupportedPrefType(prefType)) {
+                    prefType = "undefined";
+                }
+            } else if (Array.isArray(prefObj)) {
```

Comment:
Well, in my observations, even nested prefs were always objects. Did you really spot a case where this was not the case?